### PR TITLE
`Adaptive learning`: Add participation view for learning paths

### PR DESCRIPTION
**Pull request by MaximilianAnzinger**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I added pre-authorization annotations according to the [guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/#rest-endpoint-best-practices-for-authorization) and checked the course groups for all new REST Calls (security).
- [x] I documented the Java code using JavaDoc style.
#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

This PR adds the participation view to allow students to progress through their learning path by accessing Lecture Units and Exercises.

### Description

This PR adds a participation view consisting of:
- (left) Learning Path Graph (placeholder for the sequenced learning path -> follow-up PR)
- (center) Lecture Unit/Exercise
- (right) Communication Section

### Steps for Testing

Prerequisites:
- 1 Students
- 1 Course with learning paths enabled
- 1 Competency that is linked to all Exercise types and Lecture Unit types

1. Log in to Artemis
2. for each exercise type
    1. click on the node in the learning path graph (left side)
    2. start and participate in the exercise, make sure everything works as expected
    3. navigate back to the learning path participation view
    4. you should see the exercise you have been working on
    5. make sure communication works as expected (right side)
3. for each lecture unit type
    1. click on the node in the learning path graph (left side)
    2. access the lecture unit
    3. make sure everything works as expected (you can download files, view embedded videos etc.)
    4. make sure communication works as expected (right sode)
4. Click on the `Previous` button the participation view should load the task you were working on prior to the last one
5. repeat 4. and make sure the order of the loaded tasks is as expected

#### Performance Review
- [x] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [x] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [learning-path.service.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5294/latest/artifact/shared/Coverage-Report-Client-Tests/app/course/learning-paths/learning-path.service.ts.html) | 68.75% | ✅ |
| [learning-path-container.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5294/latest/artifact/shared/Coverage-Report-Client-Tests/app/course/learning-paths/participate/learning-path-container.component.ts.html) | 93.5% | ✅ |
| [learning-path-graph-sidebar.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5294/latest/artifact/shared/Coverage-Report-Client-Tests/app/course/learning-paths/participate/learning-path-graph-sidebar.component.ts.html) | 68.75% | ✅ |
| [learning-path-history-storage.service.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5294/latest/artifact/shared/Coverage-Report-Client-Tests/app/course/learning-paths/participate/learning-path-history-storage.service.ts.html) | 95.83% | ✅ |
| [learning-path-lecture-unit-view.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5294/latest/artifact/shared/Coverage-Report-Client-Tests/app/course/learning-paths/participate/lecture-unit/learning-path-lecture-unit-view.component.ts.html) | 90.47% | ✅ |
| [course-overview.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5294/latest/artifact/shared/Coverage-Report-Client-Tests/app/overview/course-overview.component.ts.html) | 95.51% | ✅ |
| [discussion-section.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5294/latest/artifact/shared/Coverage-Report-Client-Tests/app/overview/discussion-section/discussion-section.component.ts.html) | 80.32% | ✅ |
| [course-exercise-details.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5294/latest/artifact/shared/Coverage-Report-Client-Tests/app/overview/exercise-details/course-exercise-details.component.ts.html) | 84.03% | ✅ |
| [navbar.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5294/latest/artifact/shared/Coverage-Report-Client-Tests/app/shared/layouts/navbar/navbar.component.ts.html) | 89.89% | ✅ |

#### Server

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [LearningPathResource.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5294/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest/LearningPathResource.html) | 92% | ✅ |

�

### Screenshots

Screenshot 2023-08-15 at 01 45 38

### CODE REVIEW

**Review by tobias-lippert — DISMISSED** [review]
Tested in testing session on ts3. Participation view works flawlessly.

**Review by JohannesStoehr — DISMISSED** [review]
Tested on TS3 during the testing session and works as expected.

**Review by laadvo — APPROVED** [review]
Tested file upload, modeling and text exercise on ts3 during the testing session, participation view worked as expected.

File: src/main/webapp/app/course/learning-paths/participate/learning-path-container.component.ts [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    faChevronRight = faChevronRight;
+
+    constructor(
+        private router: Router,
+        private activatedRoute: ActivatedRoute,
+        private alertService: AlertService,
+        private learningPathService: LearningPathService,
+        private lectureService: LectureService,
+        private exerciseService: ExerciseService,
+        public learningPathHistoryStorageService: LearningPathHistoryStorageService,
```
Comment (tobias-lippert) [inline src/main/webapp/app/course/learning-paths/participate/learning-path-container.component.ts:45]: Why is this service not private?

File: src/main/webapp/app/course/learning-paths/participate/learning-path-container.component.ts [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        });
+    }
+
+    onNextTask() {
+        if (this.lectureUnit?.id) {
+            this.learningPathHistoryStorageService.storeLectureUnit(this.learningPathId, this.lectureId!, this.lectureUnit.id);
+        } else if (this.exercise?.id) {
+            this.learningPathHistoryStorageService.storeExercise(this.learningPathId, this.exercise.id);
+        }
+        this.undefineAll();
```
Comment (tobias-lippert) [inline src/main/webapp/app/course/learning-paths/participate/learning-path-container.component.ts:68]: Why is this necessary?

File: src/main/webapp/app/course/learning-paths/participate/learning-path-graph-sidebar.component.html [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+                </div>
+            </div>
+            <!-- body -->
+            <div class="card-body">
+                <jhi-learning-path-graph
+                    #learningPathGraphComponent
+                    *ngIf="learningPathId"
+                    [courseId]="courseId"
+                    [learningPathId]="learningPathId"
+                    (nodeClicked)="nodeClicked.emit($event)"
```
Comment (tobias-lippert) [inline src/main/webapp/app/course/learning-paths/participate/learning-path-graph-sidebar.component.html:21]: Why do you emit a on click event if a on click event is emitted?

File: src/main/webapp/app/course/learning-paths/participate/learning-path-graph-sidebar.component.scss [hunk 4]
```diff
@@ -0,0 +1,77 @@
+@import 'src/main/webapp/content/scss/artemis-variables';
+
+$draggable-width: 15px;
+$graph-min-width: 215px;
+
+.learning-path-sidebar {
+    .expanded-graph {
+        display: flex;
+        width: calc(#{$draggable-width} + #{$graph-min-width});
+        min-height: 500px;
```
Comment (tobias-lippert) [inline src/main/webapp/app/course/learning-paths/participate/learning-path-graph-sidebar.component.scss:10]: Are learning paths intended to be viewed on mobile devices?

File: src/main/webapp/app/course/learning-paths/participate/learning-path-graph-sidebar.component.scss [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+            .card-body {
+                padding: 0;
+            }
+        }
+    }
+
+    .collapsed-graph {
+        display: flex;
+        width: 38px;
```
Comment (tobias-lippert) [inline src/main/webapp/app/course/learning-paths/participate/learning-path-graph-sidebar.component.scss:50]: Out of curiosity: why 38?

File: src/main/webapp/app/overview/exercise-details/course-exercise-details.component.ts [hunk 6]
```diff
@@ -68,8 +68,9 @@ export class CourseExerciseDetailsComponent implements OnInit, OnDestroy {
     readonly isCommunicationEnabled = isCommunicationEnabled;
     readonly isMessagingEnabled = isMessagingEnabled;
 
+    public learningPathMode = false;
     private currentUser: User;
-    private exerciseId: number;
+    public exerciseId: number;
```
Comment (tobias-lippert) [inline src/main/webapp/app/overview/exercise-details/course-exercise-details.component.ts:73]: Why do we need to make this public?

File: src/test/javascript/spec/component/learning-paths/participate/learning-path-lecture-unit-view.component.spec.ts [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+                lectureUnitService = TestBed.inject(LectureUnitService);
+                setCompletionStub = jest.spyOn(lectureUnitService, 'setCompletion');
+            });
+    });
+
+    afterEach(() => {
+        jest.restoreAllMocks();
+    });
+
+    it('should display attachment unit correctly', () => {
```
Comment (tobias-lippert) [inline src/test/javascript/spec/component/learning-paths/participate/learning-path-lecture-unit-view.component.spec.ts:42]: Have you thought about using a parameterized test instead of all the different ones?

File: src/main/webapp/i18n/en/competency.json [hunk 8]
```diff
@@ -174,6 +178,18 @@
...
                 }
+            },
+            "sideBar": {
+                "hide": "Hide Learning Path",
+                "show": "Show Learning Path",
+                "header": "Learning Path"
+            },
+            "participate": {
+                "noTaskSelected": "Currently you have no lecture unit or exercise selected.",
+                "prev": "Previous",
```
Comment (tobias-lippert) [inline src/main/webapp/i18n/en/competency.json:189]: Semantics of German and English translation differ again IMO

File: src/main/webapp/app/overview/discussion-section/discussion-section.component.ts [hunk 9]
```diff
@@ -70,7 +70,11 @@ export class DiscussionSectionComponent extends CourseDiscussionDirective implem
             this.course = this.exercise?.course ?? this.lecture?.course;
             this.metisService.setCourse(this.course);
             this.metisService.setPageType(this.pageType);
-            this.setChannel(routeParams.params.courseId);
+            if (routeParams.params.courseId) {
```
Comment (tobias-lippert) [inline src/main/webapp/app/overview/discussion-section/discussion-section.component.ts:73]: Why do we need to distinguish theses cases now?

File: src/main/webapp/app/course/learning-paths/participate/lecture-unit/learning-path-lecture-unit-view.module.ts [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+
+const routes: Routes = [
+    {
+        path: '',
+        component: LearningPathLectureUnitViewComponent,
+        data: {
+            authorities: [Authority.USER],
+            pageTitle: 'overview.learningPath',
+        },
+        canActivate: [UserRouteAccessService],
```
Comment (tobias-lippert) [inline src/main/webapp/app/course/learning-paths/participate/lecture-unit/learning-path-lecture-unit-view.module.ts:17]: Be aware that you do not always use this service in your newly introduced routes (in another file).

**Review by tobias-lippert — COMMENTED** [review]
Looks goo, but as always I have a few questions.

**Review by nityanandaz — APPROVED** [review]
Tested on TS3 during testing session (2023-08-15), code looks neat

File: src/main/java/de/tum/in/www1/artemis/web/rest/LearningPathResource.java [hunk 11]
```diff
@@ -165,4 +165,35 @@ else if (!authorizationCheckService.isAtLeastInstructorInCourse(course, user) &&
...
     }
+
+    /**
+     * GET /courses/:courseId/learning-path-id : Gets the id of the learning path.
+     * If the learning path has not been generated although the course has learning paths enabled, the corresponding learning path will be created.
+     *
+     * @param courseId the id of the course from which the learning path id should be fetched
+     * @return the ResponseEntity with status 200 (OK) and with body the id of the learning path
+     */
+    @GetMapping("/courses/{courseId}/learning-path-id")
```
[the remark this answers is not in the document]
Comment (MaximilianAnzinger) [reply src/main/java/de/tum/in/www1/artemis/web/rest/LearningPathResource.java:176]: I asked my self the same question regarding the GET/PUT thing. In 99.9% the learning path should already be created by the instructor (when enabling learning paths) or during enrollment. This endpoint will only create it if someone was added to the group outside of the course context (e.g. student was manually added to the group in Jira). Rather than displaying an error/a prompt to contact your instructor, I decided to generate the LP so student's have a better experience.

> And why do we only return an ID?

For now, everything (competencies, lectures, exercises, etc.) is strictly owned by a course. In the future, we are aiming for standardized competencies that are course independent. Furthermore, currently, a learning path is generated for a specific course. This will be extended to Artemis-wide learning paths in the future (one user has exactly one learning path). The design of the endpoints for learning paths already incorporates this vision (as much as possible). Hence, all requests (sent by a user after requesting the LP-id) work with the LP-id to avoid major changes down the line.

File: src/main/webapp/app/course/learning-paths/participate/learning-path-graph-sidebar.component.html (hunk 3 above)
Comment (MaximilianAnzinger) [reply src/main/webapp/app/course/learning-paths/participate/learning-path-graph-sidebar.component.html:21]: This just passes the event (received from the child) to the parent.

File: src/main/webapp/app/course/learning-paths/participate/learning-path-graph-sidebar.component.scss (hunk 4 above)
Comment (MaximilianAnzinger) [reply src/main/webapp/app/course/learning-paths/participate/learning-path-graph-sidebar.component.scss:10]: Participating in LPs on a mobile device is not planned. If it's deemed necessary further down the line I would address this in a separate PR.

File: src/main/webapp/app/overview/exercise-details/course-exercise-details.component.ts (hunk 6 above)
Comment (MaximilianAnzinger) [reply src/main/webapp/app/overview/exercise-details/course-exercise-details.component.ts:73]: I have to set it in `LearningPathContainerComponent` when the exercise details get routed (similar to what we do for conversations). Otherwise, the exercisId would be taken from the URL which is not possible since we don't actually want to navigate to a sub-route.

File: src/main/webapp/app/overview/discussion-section/discussion-section.component.ts (hunk 9 above)
Comment (MaximilianAnzinger) [reply src/main/webapp/app/overview/discussion-section/discussion-section.component.ts:73]: Since for the learning path view the courseId is not part of the URL but set by the parent container.

File: src/main/webapp/app/course/learning-paths/participate/learning-path-container.component.ts (hunk 1 above)
Comment (MaximilianAnzinger) [reply src/main/webapp/app/course/learning-paths/participate/learning-path-container.component.ts:45]: Otherwise the service could not be referenced from the component's template.

**Review by tobias-lippert — DISMISSED** [review]
Code looks good to me now. Thanks for addressing my comments.

**Review by JohannesStoehr — APPROVED** [review]
Reapprove after changes for Tobias

**Review by laadvo — APPROVED** [review]
reapprove test

**Review by nityanandaz — APPROVED** [review]

File: src/main/webapp/app/course/learning-paths/participate/learning-path-container.component.html [hunk 12]
```diff
@@ -0,0 +1,24 @@
+<div class="row mt-2 main-container">
+    <div class="col d-flex flex-grow-1 justify-end px-0 me-1" style="max-width: min-content">
+        <jhi-learning-path-graph-sidebar
+            *ngIf="learningPathId"
+            [courseId]="courseId"
+            [learningPathId]="learningPathId"
+            (nodeClicked)="onNodeClicked($event)"
+        ></jhi-learning-path-graph-sidebar>
+    </div>
+    <div class="col pe-0 flex-grow-1" style="min-width: 200px">
```
Comment (bassner) [inline src/main/webapp/app/course/learning-paths/participate/learning-path-container.component.html:10]: Lets' avoid inline styles and move this to the `.scss` file instead.

File: src/main/webapp/app/course/learning-paths/participate/learning-path-container.component.html [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+        <div class="card" style="border-style: none">
```
Comment (bassner) [inline src/main/webapp/app/course/learning-paths/participate/learning-path-container.component.html:11]: Lets' avoid inline styles and move this to the `.scss` file instead.

File: src/main/webapp/app/course/learning-paths/participate/learning-path-container.component.scss [hunk 14]
```diff
@@ -0,0 +1,13 @@
...
+}
+
+.course-info-bar {
+    margin: 0 !important;
+}
+
+.tab-bar-exercise-details {
+    margin-left: 0 !important;
+    margin-right: 0 !important;
+}
```
Comment (bassner) [inline src/main/webapp/app/course/learning-paths/participate/learning-path-container.component.scss:13]: Hmm ... I'd like to discuss this approach - seems like you're changing the style of different components here (outside of and not even contained by this component), right?
This may lead to confusion if someone just look at the code of the affected components

File: src/main/webapp/app/course/learning-paths/participate/learning-path-container.component.ts [hunk 15]
```diff
@@ -0,0 +1,164 @@
...
+})
+export class LearningPathContainerComponent implements OnInit {
+    @Input() courseId: number;
+    learningPathId: number;
+
+    learningObjectId: number;
+    lectureId?: number;
+    lecture: Lecture | undefined;
+    lectureUnit: LectureUnit | undefined;
+    exercise: Exercise | undefined;
```
Comment (bassner) [inline src/main/webapp/app/course/learning-paths/participate/learning-path-container.component.ts:32]: Can we keep these delcarations consistent?

```suggestion
    lecture?: Lecture;
    lectureUnit?: LectureUnit;
    exercise?: Exercise;
```

File: src/main/webapp/app/course/learning-paths/participate/learning-path-container.component.ts [hunk 16]
```diff
@@ -0,0 +1,164 @@
...
+    setupExerciseView(instance: CourseExerciseDetailsComponent) {
+        if (this.exercise) {
+            instance.learningPathMode = true;
+            instance.courseId = this.courseId;
+            instance.exerciseId = this.learningObjectId;
+        }
+    }
+
+    onNodeClicked(node: NgxLearningPathNode) {
+        if (node.type === NodeType.LECTURE_UNIT || node.type === NodeType.EXERCISE) {
```
Comment (bassner) [inline src/main/webapp/app/course/learning-paths/participate/learning-path-container.component.ts:147]: Let's return early to avoid nesting the entire function

```suggestion
        if (node.type !== NodeType.LECTURE_UNIT && node.type !== NodeType.EXERCISE) {
            return;
        }
```

File: src/main/webapp/app/course/learning-paths/participate/learning-path-graph-sidebar.component.ts [hunk 17]
```diff
@@ -0,0 +1,53 @@
...
+@Component({
+    selector: 'jhi-learning-path-graph-sidebar',
+    styleUrls: ['./learning-path-graph-sidebar.component.scss'],
+    templateUrl: './learning-path-graph-sidebar.component.html',
+})
+export class LearningPathGraphSidebarComponent implements AfterViewInit {
+    @Input() courseId: number;
+    @Input() learningPathId: number;
+    collapsed: boolean;
+    // Icons
```
Comment (bassner) [inline src/main/webapp/app/course/learning-paths/participate/learning-path-graph-sidebar.component.ts:16]:
```suggestion

    // Icons
```

File: src/main/webapp/app/course/learning-paths/participate/learning-path-graph-sidebar.component.ts [hunk 18]
```diff
@@ -0,0 +1,53 @@
...
+    faChevronLeft = faChevronLeft;
+    faChevronRight = faChevronRight;
+    faGripLinesVertical = faGripLinesVertical;
+    faNetworkWired = faNetworkWired;
+
+    @ViewChild(`learningPathGraphComponent`, { static: false })
```
Comment (bassner) [inline src/main/webapp/app/course/learning-paths/participate/learning-path-graph-sidebar.component.ts:22]:
```suggestion
    @ViewChild('learningPathGraphComponent', { static: false })
```

File: src/main/webapp/app/course/learning-paths/participate/learning-path-history-storage.service.ts [hunk 19]
```diff
@@ -0,0 +1,85 @@
...
+    /**
+     * Stores the exercise in the learning path's history.
+     *
+     * @param learningPathId the id of the learning path to which the new entry should be added
+     * @param exerciseId the id of the exercise
+     */
+    storeExercise(learningPathId: number, exerciseId: number) {
+        this.store(learningPathId, new ExerciseEntry(exerciseId));
+    }
+    private store(learningPathId: number, entry: HistoryEntry) {
```
Comment (bassner) [inline src/main/webapp/app/course/learning-paths/participate/learning-path-history-storage.service.ts:30]:
```suggestion

    private store(learningPathId: number, entry: HistoryEntry) {
```

File: src/main/webapp/app/course/learning-paths/participate/learning-path-history-storage.service.ts [hunk 20]
```diff
@@ -0,0 +1,85 @@
...
+    }
+
+    /**
+     * Gets and removes the latest stored entry from the learning path's history.
+     *
+     * @param learningPathId
+     */
+    getPrevious(learningPathId: number) {
+        if (!this.hasPrevious(learningPathId)) {
+            return null;
```
Comment (bassner) [inline src/main/webapp/app/course/learning-paths/participate/learning-path-history-storage.service.ts:59]: Can we [use `undefined` instead](https://ls1intum.github.io/Artemis/dev/guidelines/client/#null-and-undefined)?

File: src/main/webapp/app/course/learning-paths/participate/lecture-unit/learning-path-lecture-unit-view.component.html [hunk 21]
```diff
@@ -0,0 +1,17 @@
+<div class="row">
+    <div class="col pe-0 flex-grow-1">
+        <div [ngSwitch]="lectureUnit.type">
+            <jhi-attachment-unit *ngSwitchCase="LectureUnitType.ATTACHMENT" [attachmentUnit]="lectureUnit" (onCompletion)="completeLectureUnit($event)"></jhi-attachment-unit>
+            <jhi-video-unit *ngSwitchCase="LectureUnitType.VIDEO" [videoUnit]="lectureUnit" (onCompletion)="completeLectureUnit($event)"></jhi-video-unit>
+            <jhi-text-unit *ngSwitchCase="LectureUnitType.TEXT" [textUnit]="lectureUnit" (onCompletion)="completeLectureUnit($event)"></jhi-text-unit>
+            <jhi-online-unit *ngSwitchCase="LectureUnitType.ONLINE" [onlineUnit]="lectureUnit" (onCompletion)="completeLectureUnit($event)"></jhi-online-unit>
+        </div>
+    </div>
+    <div class="col d-flex flex-grow-1 justify-end" style="max-width: min-content">
```
Comment (bassner) [inline src/main/webapp/app/course/learning-paths/participate/lecture-unit/learning-path-lecture-unit-view.component.html:10]: Lets' avoid inline styles and move this to the .scss file instead.

File: src/main/webapp/app/course/learning-paths/participate/lecture-unit/learning-path-lecture-unit-view.component.ts [hunk 22]
```diff
@@ -0,0 +1,58 @@
...
+                next: () => {
+                    event.lectureUnit.completed = event.completed;
+                },
+                error: (res: HttpErrorResponse) => onError(this.alertService, res),
+            });
+        }
+    }
+
+    protected readonly isMessagingEnabled = isMessagingEnabled;
+    protected readonly isCommunicationEnabled = isCommunicationEnabled;
```
Comment (bassner) [inline src/main/webapp/app/course/learning-paths/participate/lecture-unit/learning-path-lecture-unit-view.component.ts:44]: We should move these fields to the top of the class, above the constructor

**Review by Strohgelaender — APPROVED** [review]
tested on ts2, works. code looks good to me as well.

**Review by YeetTheFirst21 — APPROVED** [review]
Tested on TS2 in the testing session

**Review by JohannesStoehr — APPROVED** [review]
Reapprove code

**Review by tobias-lippert — APPROVED** [review]
Reapprove

**Review by bassner — APPROVED** [review]
